editor/editor_run.cpp

```diff
@@ -158,17 +158,40 @@ Error EditorRun::run(const String &p_scene, const String &p_write_movie) {
 
 	int window_placement = EDITOR_GET("run/window_placement/rect");
 	if (screen_rect != Rect2()) {
-		Size2 window_size;
-		window_size.x = GLOBAL_GET("display/window/size/viewport_width");
-		window_size.y = GLOBAL_GET("display/window/size/viewport_height");
-
-		Size2 desired_size;
-		desired_size.x = GLOBAL_GET("display/window/size/window_width_override");
-		desired_size.y = GLOBAL_GET("display/window/size/window_height_override");
-		if (desired_size.x > 0 && desired_size.y > 0) {
-			window_size = desired_size;
+		const int window_width_override = GLOBAL_GET("display/window/size/window_width_override");
+		const int window_height_override = GLOBAL_GET("display/window/size/window_height_override");
+
+		const int viewport_width = GLOBAL_GET("display/window/size/viewport_width");
+		const int viewport_height = GLOBAL_GET("display/window/size/viewport_height");
+
+		const int window_minimum_width = GLOBAL_GET("display/window/size/window_minimum_width");
+		const int window_minimum_height = GLOBAL_GET("display/window/size/window_minimum_height");
+		const int window_maximum_width = GLOBAL_GET("display/window/size/window_maximum_width");
+		const int window_maximum_height = GLOBAL_GET("display/window/size/window_maximum_height");
+
+		int desired_width = viewport_width;
+		int desired_height = viewport_height;
+
+		if (window_width_override != 0) {
+			desired_width = window_width_override;
+		}
+		if (window_height_override != 0) {
+			desired_height = window_height_override;
 		}
 
+		desired_width = std::max(desired_width, window_minimum_width);
+		desired_height = std::max(desired_height, window_minimum_height);
+		if (window_maximum_width != 0) {
+			desired_width = std::min(desired_width, window_maximum_width);
+		}
+		if (window_maximum_height != 0) {
+			desired_height = std::min(desired_height, window_maximum_height);
+		}
```

Comment:
I didn't test the code, but how would this behave if `window_minimum_width` is greater than `window_maximum_width`? Sounds like it would shrink the window width to `window_maximum_width`, not respecting `window_minimum_width` anymore.

Reply:
You could probably simply use `CLAMP` for this.